services/motion/builtin/move_request.go

```diff
@@ -164,6 +192,72 @@ func (mr *moveRequest) deviatedFromPlan(ctx context.Context, plan motionplan.Pla
 	return state.ExecuteResponse{}, nil
 }
 
+// getTransientDetections returns a list of geometries as observed by the provided vision service and camera.
+// Depending on the caller, the geometries returned are either in their relative position
+// with respect to the base or in their absolute position with respect to the world.
+func (mr *moveRequest) getTransientDetections(
+	ctx context.Context,
+	visSrvc vision.Service,
+	camName resource.Name,
+) (*referenceframe.GeometriesInFrame, error) {
+	mr.logger.CDebugf(ctx,
+		"proceeding to get detections from vision service: %s with camera: %s",
+		visSrvc.Name().ShortName(),
+		camName.ShortName(),
+	)
+	// any obstacles specified by the worldstate of the moveRequest will also re-detected here.
+	// get detections from vision service
+	detections, err := visSrvc.GetObjectPointClouds(ctx, camName.Name, nil)
+	if err != nil {
+		return nil, err
+	}
+	mr.logger.CDebugf(ctx, "got %d detections", len(detections))
+
+	if len(detections) == 0 {
+		mr.logger.CDebug(ctx, "no detections, returning early")
+		return referenceframe.NewGeometriesInFrame(referenceframe.World, []spatialmath.Geometry{}), nil
+	}
+
+	// determine transform of camera to base
+	cameraOrigin := referenceframe.NewPoseInFrame(camName.ShortName(), spatialmath.NewZeroPose())
+	cameraToBase, err := mr.fsService.TransformPose(ctx, cameraOrigin, mr.kinematicBase.Name().ShortName(), nil)
+	if err != nil {
+		// here we make the assumption the camera is coincident with the base
+		mr.logger.CDebugf(ctx,
+			"we assume the base named: %s is coincident with the camera named: %s due to err: %v",
+			mr.kinematicBase.Name().ShortName(), camName.ShortName(), err.Error(),
+		)
+		cameraToBase = cameraOrigin
+	}
+	mr.logger.CDebugf(ctx, "cameraToBase transform: %v", spatialmath.PoseToProtobuf(cameraToBase.Pose()))
+
+	// detections placed in the base frame
+	relativeGeoms := []spatialmath.Geometry{}
+	for i, detection := range detections {
+		geometry := detection.Geometry
+		// update the label of the geometry so we know it is transient
+		label := camName.ShortName() + "_transientObstacle_" + strconv.Itoa(i)
+		if geometry.Label() != "" {
+			label += "_" + geometry.Label()
+		}
+		geometry.SetLabel(label)
+		mr.logger.CDebugf(ctx, "detection %d observed from the camera frame coordinate system: %s - %s",
+			i, camName.ShortName(), geometry.String(),
+		)
+
+		// transform the transform the geometry to be relative to the base frame which is +Y forwards
+		relativeGeom := geometry.Transform(cameraToBase.Pose())
+		mr.logger.CDebugf(ctx, "detection %d observed from the camera's position in the base frame coordinate system has pose: %v",
+			i, spatialmath.PoseToProtobuf(relativeGeom.Pose()),
+		)
+		relativeGeoms = append(relativeGeoms, relativeGeom)
+	}
+	return referenceframe.NewGeometriesInFrame(referenceframe.World, relativeGeoms), nil
```

Comment:
I agree that we do transform the geometries to be in the frame of the base, but they should not be interpreted that way.

Within `relativeMoveRequestFromAbsolute` we already also transform into the base frame but do not treat it as such.
https://github.com/viamrobotics/rdk/blob/main/services/motion/builtin/move_request.go#L664
```
	// convert GeoObstacles into GeometriesInFrame with respect to the base's starting point
	geoms := make([]spatialmath.Geometry, 0, len(worldObstacles))
	for _, geom := range worldObstacles {
		geoms = append(geoms, geom.Transform(startPoseInv))
	}

	gif := referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)
```

Reply:
`relativeMoveRequestFromAbsolute` starts by creating the `baseOnlyFs` by subsetting the frame system, creating a new frame system where the base is guaranteed to be coincident with `World`. Within the scope of that function, base and world are guaranteed to be coincident.

Since we don't enforce that all `moveRequest`s must have the base coincident with `World`, this isn't necessarily the case here.